aten/src/ATen/native/TensorShape.cpp

```diff
@@ -622,6 +622,10 @@ std::vector<Tensor> meshgrid(TensorList tensors) {
       AT_ERROR("Expected scalar or 1D tensor in the tensor list but got: ", tensors[i]);
     }
   }
+  for(int64_t i = 0; i < size - 1; i++){
+      AT_CHECK(tensors[i].dtype()==tensors[i+1].dtype(),"meshgrid expects all tensors to have the same dtype");
+      AT_CHECK(tensors[i].device()==tensors[i+1].device(),"meshgrid expects all tensors to have the same device");
```

Comment:
Perhaps here we could have made use of the `checkAllSameType` and `checkAllSameGPU` functions in `TensorUtils.h` here, but those are defined for a list of `TensorArg`s instead of `TensorList`. Should they be overloaded? @zou3519 @SsnL